region/norfair/east/Green Bubbles Missile Room.json

```diff
@@ -87,6 +88,46 @@
         "h_canHeatedCrystalFlash"
       ]
     },
+    {
+      "link": [1, 2],
+      "name": "Green Bubbles Shinespark",
+      "notable": true,
+      "entranceCondition": {
+        "comeInShinecharging": {
+          "length": 3,
+          "openEnd": 0
+        }
+      },
+      "requires": [
+        "canShinechargeMovementTricky",
+        "canMockball",
+        {"enemyDamage": {
+          "enemy": "Geruta",
+          "type": "contact",
+          "hits": 1
+        }},
+        {"heatFrames": 200},
+        {"shinespark": {"frames": 11, "excessFrames": 0}}
+      ],
+      "exitCondition": {
+        "leaveWithSpark": {
+          "position": "bottom"
+        }
+      },
+      "unlocksDoors": [
+        {"types": ["super"], "requires": []},
+        {"types": ["missiles", "powerbomb"], "requires": ["never"]}
+      ],
+      "note": [
+        "Fire a shot on the way up to clear the shot block.",
+        "Perform a mockball, tank a hit from the Geruta, and unmorph while landing on the bubble pillar.",
+        "Fire a shot to open the door, and activate the shinespark wind-up.",
+        "Wait to spark out until after the door has finished opening."
+      ],
+      "devNote": [
+        "FIXME: Figure out if a top-position spark might be possible."
+      ]
+    },
```

Comment:
```suggestion
      "link": [1, 2],
      "name": "Tricky Movement Shinespark",
      "entranceCondition": {
        "comeInShinecharging": {
          "length": 3,
          "openEnd": 0
        }
      },
      "requires": [
        "canShinechargeMovementTricky",
        "canMidairShinespark",
        "canMockball",
        {"or": [
          {"enemyKill": {
            "enemies": [["Geruta"]],
            "explicitWeapons": [
              "Missile",
              "Super",
              "Wave",
              "Spazer",
              "Plasma"
            ]
          }},
          {"enemyDamage": {
            "enemy": "Geruta",
            "type": "contact",
            "hits": 1
          }}
        ]},
        {"heatFrames": 200},
        {"shinespark": {"frames": 16, "excessFrames": 0}}
      ],
      "exitCondition": {
        "leaveWithSpark": {}
      },
      "unlocksDoors": [
        {"types": ["super"], "requires": []},
        {"types": ["missiles", "powerbomb"], "requires": ["never"]}
      ],
      "note": [
        "Fire a shot on the way up to clear the shot block.",
        "Perform a mockball but unmorph after exiting the Morph hole and Kill the Geruta while falling, if possible.",
        "Fire a shot to open the door, then jump forward and Shinespark out."
      ]
    },
```

Reply:
For landing on the bubble pillar, Kago unmorphing makes it pretty easy.
But you can reach the door and spark from there with either position too.  So maybe you weren't 1)releasing run on the first jump, 2)getting enough run speed, or 3)jumping late for the mockball.

Reply:
It could just be because this is how I always try to do the room normally, but this doesn't feel notable to me.